dapps/marketplace/src/pages/identity/OAuthAttestation.js

```diff
@@ -169,35 +184,81 @@ class OAuthAttestation extends Component {
     )
   }
 
+  renderAppDownloadButton() {
+    return (
+      <>
+        <a
+          className="btn btn-primary"
+          href={
+            this.state.mobilePlatform === 'ios'
+              ? 'https://itunes.apple.com/us/app/apple-store/id1446091928?mt=8'
+              : 'https://originprotocol.com/mobile'
+          }
+        >
+          <fbt desc="OAuthAttestation.getAppButton">Get the Origin App</fbt>
+        </a>
+      </>
+    )
+  }
+
   renderGenerateCode({ authUrl, redirect }) {
     const { isMobile } = this.props
+    const { isForbidden } = this.state
     const providerName = getProviderDisplayName(this.props.provider)
 
-    const header = isMobile ? (
-      <fbt desc="OAuthAttestation.tapToBegin">
-        Tap the button below to begin.
-      </fbt>
-    ) : (
-      <fbt desc="OAuthAttestation.verify">
-        Verify your <fbt:param name="provider">{providerName}</fbt:param>{' '}
-        Account
-      </fbt>
-    )
+    const header =
+      isMobile && !isForbidden ? (
+        <fbt desc="OAuthAttestation.tapToBegin">
+          Tap the button below to begin
+        </fbt>
+      ) : isMobile && isForbidden ? (
+        <fbt desc="OAuthAttestation.getAppHeader">
+          Get the Origin Marketplace App
+        </fbt>
+      ) : (
+        <fbt desc="OAuthAttestation.verify">
+          Verify your <fbt:param name="provider">{providerName}</fbt:param>{' '}
+          Account
+        </fbt>
+      )
 
     return (
       <>
         <h2>{header}</h2>
-        <div className="help mt-0 mb-3">
-          <fbt desc="OAuthAttestation.neverPost">
-            We will never post on your behalf.
-          </fbt>
-        </div>
+        {!isForbidden && (
```

Comment:
Adding to my previous comment, I feel it'd be better if we move Forbidden modal to a separate function or make it a separate component that can used in other places.

Reply:
Yes, and it's probably better to separate concerns anyway.

Reply:
I agree that a separate component makes a lot of sense. I'll go and separate concerns.